virttest/qemu_vm.py

```diff
@@ -2785,8 +2785,14 @@ def create(self, name=None, params=None, root_dir=None,
             elif migration_mode == "exec":
                 if migration_exec_cmd is None:
                     self.migration_port = utils_misc.find_free_port(5200, 6000)
-                    qemu_command += (' -incoming "exec:nc -l %s"' %
-                                     self.migration_port)
+                    # check whether ip version supported by nc
+                    if process.system("nc -h | grep -E '\-4 | \-6'") == 0:
```

Comment:
There are 2 problems to this statement:

1. using shell features (pipes `|`) without passing `shell=True`, the command would never be executed correctly
```
# cat /tmp/fakenc 
#!/bin/bash
cat << EOF
Ncat 7.70 ( https://nmap.org/ncat )
Usage: ncat [options] [hostname] [port]

Options taking a time assume seconds. Append 'ms' for milliseconds,
's' for seconds, 'm' for minutes, or 'h' for hours (e.g. 500ms).
  -h, --help                 Display this help screen

See the ncat(1) manpage for full options, descriptions and usage examples
EOF
```
```
# python3
Python 3.6.8 (default, Jul  3 2019, 10:59:07) 
[GCC 8.3.1 20190507 (Red Hat 8.3.1-4)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from avocado.utils import process
>>> process.system("nc -h | grep -E '\-4 | \-6'")
0
>>> process.system("/tmp/fakenc -h | grep -E '\-4 | \-6'")
0
```

2. missing the argument `ignore_status=True`, it will never jump into the `else` clause
```
>>> process.system("nc -h | grep -E '\-4 | \-6'", shell=True)
0
>>> process.system("/tmp/fakenc -h | grep -E '\-4 | \-6'", shell=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/kar/workspace/avocado/avocado/utils/process.py", line 1101, in system
    encoding=encoding)
  File "/home/kar/workspace/avocado/avocado/utils/process.py", line 1038, in run
    raise CmdError(cmd, sp.result)
avocado.utils.process.CmdError: Command '/tmp/fakenc -h | grep -E '\-4 | \-6'' failed.
stdout: b''
stderr: b''
additional_info: None
```

So, next time please do branch coverage and check the result carefully.

Reply:
Thank you for taking care explaining things. I will ensure considering it going forward.